### DISCUSSION TITLE
Generic way to check user permissions

### DISCUSSION THREAD

**Discussion by matthias-hiemer** [eadwinCode/django-ninja-extra#50] [Q&A]
Hi there! 

I'm new to Ninja Extra and was wondering how I can check for permissions in a generic way.
Checking a single permission class of type BasePermission works fine to protect an endpoint.

For Example:
```
    class HasDocumentPermission(permissions.BasePermission):

        def has_permission(self, request, controller) -> bool:
            user = request.user or request.auth
            return user and user.has_perm("api.view_document")

    @http_get(
        "",
        response={200: List[DocumentSchema], 404: Message},
        permissions=[HasDocumentPermission],
        by_alias=True,
    )
    @requires()
    def get_documents(self):
        # Gets documents
```

But: We have multiple permissions in multiple groups in our project. So, creating boilerplate permission classes doesn't seem to be the most elegant way. 
Can I do this somehow by using a BasePermission class, that takes a permission as parameter? Or do I need another concept, like another decorator?

Thanks and best regards
Matthias

**Comment by lewoudar** [comment]
Hi @matthias-hiemer it seems like in the latest [release](https://github.com/eadwinCode/django-ninja-extra/commit/388023df3e462e7dfce23a8abbd603e2d1d32cae), you are now able to use the BasePermission class. Hope it will help you.
